### PR TITLE
Order History events in the same confirmation height.

### PR DESCRIPTION
**Pull request by junderw**
# REQUIRES RE-INDEX

# REQUIRES A LOT OF TESTING

This should fix the ordering of history entries and transactions within the same block/tx.

I think the addition of this data to the rows might cause unforseen consequences... so it needs to be tested thoroughly.

Requires a Re-Index.

### CODE REVIEW

**Comment by junderw** [comment]
(Included #92 because `mempool` branch is broken without it)

**Comment by junderw** [comment]
The behavior of the RocksDB is relied upon in weird ways, so the biggest worry I have is a re-surfacing of the weird balance-is-off bugs...

File: src/new_index/schema.rs [hunk 1]
```diff
@@ -1602,6 +1616,7 @@ pub struct TxHistoryKey {
     pub code: u8,              // H for script history or I for asset history (elements only)
     pub hash: FullHash, // either a scripthash (always on bitcoin) or an asset id (elements only)
     pub confirmed_height: u32, // MUST be serialized as big-endian (for correct scans).
+    pub tx_position: u32, // MUST be serialized as big-endian (for correct scans). Position in block.
```
Comment (mononaut) [inline src/new_index/schema.rs]: does this need to be a u32? unlike transaction outputs, the total number of transactions in a block is consensus-bounded well below 2^16.

I suppose it only saves a small % of the total key size, but that still adds up to a few dozen GB on mainnet 😄

**Review by mononaut — APPROVED** [review]
Tested ACK @ [5bee69f7]

I tested a full reindex on testnet4 and signet without any hiccups, and confirmed that same-block transaction history is now properly ordered.

The address history summary functions need a few changes to take advantage of the new schema, but I can add that in a follow up PR.

I also stepped through all of the code that interacts with the tx history database, and couldn't see any obvious problems downstream of the schema change.

(I have a mainnet reindex still in progress, but it hasn't crashed yet!)

File: src/new_index/schema.rs (hunk 1 above)
Comment (junderw) [reply src/new_index/schema.rs]: Changed to u16

**Comment by mononaut** [comment]
(re)tested ACK @ [3dab4c7e]

I completed a full mainnet reindex with the new u16 tx position, and everything looks good so far, including balances on a bunch of heavily-trafficked addresses.

**Pull request merged**